base/reduce.jl

```diff
@@ -186,7 +186,7 @@ sum_pairwise_blocksize(::Abs2Fun) = 4096
 mapreduce_impl(f, op::AddFun, A::AbstractArray, ifirst::Int, ilast::Int) =
     mapreduce_pairwise_impl(f, op, A, ifirst, ilast, sum_pairwise_blocksize(f))
 
-sum(f::Union{Callable,Func{1}}, a) = mapreduce(f, AddFun(), a)
+sum(f, a) = mapreduce(f, AddFun(), a)
```

Comment:
This seems weird to me, given the existence of `sum(array, dims)`. I think ideally this definition would not exist at all.
`count` is ok, since its first argument is always a function.